src/sqlfluff/core/templaters.py

```diff
@@ -377,3 +379,155 @@ def process(self, in_str, fname=None, config=None):
                 )
             )
             return None, violations
+
+
+@dataclass
+class DbtConfigArgs:
+    """Arguments to load Dbt runtime config."""
+
+    project_dir: Optional[str] = None
+    profiles_dir: Optional[str] = None
+    profile: Optional[str] = None
+
+
+@register_templater
+class DbtTemplateInterface(PythonTemplateInterface):
+    """A templater using DBT."""
+
+    name = "dbt"
+
+    def __init__(self, **kwargs):
+        self.dbt_config = None
+        self.dbt_compiler = None
+        self.dbt_manifest = None
+        self.dbt_selector_method = None
+        super(DbtTemplateInterface).__init__(**kwargs)
+
+    def load_dbt_config(self, config):
+        """Loads the dbt config."""
+        from dbt.config.runtime import RuntimeConfig as DbtRuntimeConfig
+        from dbt.adapters.factory import register_adapter
+
+        self.dbt_config = DbtRuntimeConfig.from_args(
+            DbtConfigArgs(
+                project_dir=self._get_project_dir(config),
+                profiles_dir=self._get_profiles_dir(config),
+                profile=self._get_profile(config),
+            )
+        )
+        register_adapter(self.dbt_config)
+        return self.dbt_config
+
+    def load_dbt_compiler(self):
+        """Loads the dbt compiler."""
+        from dbt.compilation import Compiler as DbtCompiler
+
+        self.dbt_compiler = DbtCompiler(self.dbt_config)
+        return self.dbt_compiler
+
+    def load_dbt_manifest(self):
+        """Loads the dbt manifest."""
+        from dbt.parser.manifest import load_macro_manifest, load_manifest
+
+        # Identity function used for macro hooks
+        def identity(x):
+            return x
+
+        dbt_macros_manifest = load_macro_manifest(self.dbt_config, macro_hook=identity)
+        self.dbt_manifest = load_manifest(
+            self.dbt_config, dbt_macros_manifest, macro_hook=identity
+        )
+        return self.dbt_manifest
+
+    def load_dbt_selector_method(self):
+        """Loads the dbt selector method."""
+        from dbt.graph.selector_methods import (
+            MethodManager as DbtSelectorMethodManager,
+            MethodName as DbtMethodName,
+        )
+
+        selector_methods_manager = DbtSelectorMethodManager(
+            self.dbt_manifest, previous_state=None
+        )
+        self.dbt_selector_method = selector_methods_manager.get_method(
+            DbtMethodName.Path, method_arguments=[]
+        )
+        return self.dbt_selector_method
+
+    def _get_profiles_dir(self, config):
+        from dbt.config.profile import PROFILES_DIR
+
+        return (
+            config.get_section((self.templater_selector, self.name, "profiles_dir"))
+            or PROFILES_DIR
+        )
+
+    def _get_project_dir(self, config):
+        return config.get_section((self.templater_selector, self.name, "project_dir"))

```

Comment:
What do you think about defaulting `project_dir` to the `cwd`?

Reply:
makes sense, that was an oversight on my end

Reply:
now it's defaulting to cwd https://github.com/sqlfluff/sqlfluff/pull/508/files#diff-0f428d02c08ceb820ea8ef225ad9684548d08a6259262743253706d01e1a5953R468